e2e/e2e_test.go

```diff
@@ -41,27 +47,89 @@ import (
 )
 
 var (
-	amount            int64 = 1000000000
-	hBarSendAmount          = hedera.HbarFromTinybar(amount)
-	hbarRemovalAmount       = hedera.HbarFromTinybar(-amount)
+	receiveAmounts = map[string]uint64{
+		constants.Hbar:  100,
+		constants.Token: 100,
+	}
+	tinyBarAmount     int64 = 1000000000
+	hBarSendAmount          = hedera.HbarFromTinybar(tinyBarAmount)
+	hbarRemovalAmount       = hedera.HbarFromTinybar(-tinyBarAmount)
+	now                     = time.Now()
 )
 
 const (
 	expectedValidatorsCount = 3
 )
 
+func Test_Ethereum_Hedera_HBAR(t *testing.T) {
+	setupEnv := setup.Load()
+	now = time.Now()
+	accountBalanceBefore := util.GetHederaAccountBalance(setupEnv.Clients.Hedera, setupEnv.HederaReceiver, t)
+
+	// 1. Submit burn transaction to the bridge contract
+	submittedTx, expectedRouterBurn := sendEthTransaction(setupEnv, constants.Hbar, t)
+
+	// 2. Validate burn event
+	expectedId := validateBurnEvent(setupEnv, submittedTx, expectedRouterBurn, t)
+
+	// 3. Validate that the tx went through and emitted the correct events
+	mirrorNodeScheduledTransaction := validateScheduledTx(setupEnv, t)
+
+	// 4. Validate that the balance of the receiver account (hedera) was changed with the correct amount
+	validateReceiverAccountBalance(setupEnv, accountBalanceBefore, constants.Hbar, t)
+
+	// 5. Prepare Expected Database Record
+	expectedBurnEventRecord := util.PrepareExpectedBurnEventRecord(
+		mirrorNodeScheduledTransaction.TransactionID,
+		int64(receiveAmounts[constants.Hbar]),
+		setupEnv.HederaReceiver,
+		expectedId)
+
+	// 6. Validate Database Record
+	verifyBurnEventRecord(setupEnv.DbValidator, expectedBurnEventRecord, t)
+}
+
+func Test_Ethereum_Hedera_Token(t *testing.T) {
+	setupEnv := setup.Load()
+	now = time.Now()
+	accountBalanceBefore := util.GetHederaAccountBalance(setupEnv.Clients.Hedera, setupEnv.HederaReceiver, t)
+
+	// 1. Submit burn transaction to the bridge contract
+	submittedTx, expectedRouterBurn := sendEthTransaction(setupEnv, constants.Token, t)
+
+	// 2. Validate burn event
+	expectedId := validateBurnEvent(setupEnv, submittedTx, expectedRouterBurn, t)
+
+	// 3. Validate that the tx went through and emitted the correct events
+	mirrorNodeScheduledTransaction := validateScheduledTx(setupEnv, t)
+
+	// 4. Validate that the balance of the receiver account (hedera) was changed with the correct amount
+	validateReceiverAccountBalance(setupEnv, accountBalanceBefore, constants.Token, t)
+
+	// 5. Prepare Expected Database Record
+	expectedBurnEventRecord := util.PrepareExpectedBurnEventRecord(
+		mirrorNodeScheduledTransaction.TransactionID,
+		int64(receiveAmounts[constants.Token]),
+		setupEnv.HederaReceiver,
+		expectedId)
+
+	// 6. Validate Database Record
+	verifyBurnEventRecord(setupEnv.DbValidator, expectedBurnEventRecord, t)
+}
+
```

Comment:
If a newly created configuration is made (no previously created transfers, etc.), these tests will always fail. Since there are two `Hedera -> Ethereum` tests, it would be better to call the `Ethereum -> Hedera` tests after them, just with smaller amounts.